### PR TITLE
Add rooms that the user has left under archived in v2 sync.

### PR DESCRIPTION
**Pull request by NegativeMjark**
Tested by matrix-org/sytest#26

### CODE REVIEW

File: synapse/handlers/sync.py [hunk 1]
```diff
@@ -156,11 +175,14 @@ def initial_sync(self, sync_config):
         )
         room_list = yield self.store.get_rooms_for_user_where_membership_is(
             user_id=sync_config.user.to_string(),
-            membership_list=[Membership.INVITE, Membership.JOIN]
+            membership_list=[
+                Membership.INVITE, Membership.JOIN, Membership.LEAVE
```
Comment (erikjohnston) [inline synapse/handlers/sync.py]: Are we ignoring bans?

Comment (NegativeMjark) [inline synapse/handlers/sync.py]: I guess we should include banned rooms.

**Comment by NegativeMjark** [comment]
@erikjohnston I've included banned rooms in the archived section of sync e3d75f5.

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**